### PR TITLE
Use new Implemented Async Methods in System.Runtime.Serialization.Xml

### LINKED ISSUE #218: Add OperationFormatter.SerializeBodyAsync
We might need to have a virtual OperationFormatter.SerializeBodyAsync that delegates to the sync SerializeBody, but allows derived OperationFormatters to implement a SerializeBodyAsync.

### PR DESCRIPTION
**Pull request by shmao**
System.Runtime.Serialization.Xml implemented some async methods per our request (see https://github.com/dotnet/corefx/issues/2502). The fix is to use those methods in our code.

Fixes #218, #221, and #222.

### CODE REVIEW

**Comment by shmao** [comment]
@roncain @iamjasonp

**Comment by roncain** [comment]
Question only: do we need to upgrade our project.json dependency on serialization for this?  That work never got done pending other changes Khoa was making that would cause a newer version to become available.

**Comment by shmao** [comment]
Yes, we should update the dependency on serialization lib for this. This is still pending on https://github.com/dotnet/wcf/pull/359. The build succeeded only because the serialization package used by the build already contains the fix.

**Comment by roncain** [comment]
@shmao -- could I ask you to update the project.json and project.lock.json files to use the new versions?  I know #341 is assigned to me, but I'm swamped with other work for RC1.  If possible, we should try to get this in for RC1 or streaming does not work on Linux

File: src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs [hunk 1]
```diff
@@ -73,15 +73,15 @@ private Stream GetStreamAndWriteStartWrapperIfNecessary(XmlDictionaryWriter writ
...
+        private async Task<Stream> GetStreamAndWriteStartWrapperIfNecessaryAsync(XmlDictionaryWriter writer, object[] parameters, object returnValue)
         {
             Stream streamValue = GetStreamValue(parameters, returnValue);
             if (streamValue == null)
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(_partName);
             if (WrapperName != null)
-                writer.WriteStartElement(null, WrapperName, WrapperNamespace);
-            writer.WriteStartElement(null, PartName, PartNamespace);
-            return Task.FromResult(streamValue);
+                await writer.WriteStartElementAsync(null, WrapperName, WrapperNamespace);
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs]: NIT: tabs vs spaces

**Comment by iamjasonp** [comment]
LGTM, subject to updating project.json and project.lock.json.

**Comment by roncain** [comment]
LGTM too, subject to the .json files.
Please be aware @mconnew will also be touching the .json files for System.Net.Sockets for a critical RC1 bug fix.  We should work together to ensure the 2 PR's work.  Matt is blocked waiting for the S.N.Sockets package to be published.

File: src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs (hunk 1 above)
Comment (shmao) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs]: This is not a tab/spaces issue. The await line above is part of the if statement.

Comment (iamjasonp) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs]: ah ok. thanks for clarifying, couldn't tell based on the Git review tools :smile:

**Comment by roncain** [comment]
Just to confirm -- is the plan to update this PR to include the new serialization version dependency?  Or is it the plan to wait for @mconnew to work around the build-tools issues?

**Comment by iamjasonp** [comment]
The build tools issues that were introduced with the generation changes are already in. It should be safe to update the project.json and project.lock.json files now.

**Comment by shmao** [comment]
I'm merging this as https://github.com/dotnet/wcf/pull/455 has updated the dependency on System.Runtime.Serialization.Xml package.

**Pull request merged**